eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -406,10 +406,12 @@ class Router(val nodeParams: NodeParams, watcher: ActorRef, initialized: Option[
                     awaiting = awaiting1)
                   // we only reprocess updates and nodes if validation succeeded
                   val d2 = reprocessUpdates.foldLeft(d1) {
-                    case (d, (u, origins)) => origins.foldLeft(d) { case (d, origin) => handle(u, origin, d) } // we reprocess the same channel_update for every origin (to preserve origin information)
+                    case (d, (u, origins)) if origins.isEmpty => handle(u, None, d)
```

Comment:
We may want to use a proper typing for origin, instead of using an ActorRef, this would remove the ambiguity here.

Reply:
Done. Feels cleaner!